### PR TITLE
raft leases: use fast path to get leadership for a model

### PR DESCRIPTION
**Pull request by babbageclunk**
## Description of change

The raft lease FSM stores leases grouped by namespace+model uuid so there's no need to get all leases and then filter them by model. This can mean a lot more work when there are a lot of models. Add a lease.Store.LeaseGroup interface method to get a group of leases, and add it to all of the store implementations.

## QA steps

* Bootstrap a controller, add two models and deploy an application with multiple units in each model.
```
juju bootstrap localhost E --build-agent --verbose --model-default='logging-config="<root>=DEBUG"' --model-default="enable-os-upgrade=false" --debug  && juju add-model m1 && juju deploy -n4 ~jameinel/ubuntu-lite && juju add-model m2 && juju deploy -n4 ~jameinel/ubuntu-lite
```
* Once all the units are ready remove unit 0 in one of the models (to see the leadership change).
```
juju remove-unit -m m1 ubuntu-lite/0
```
* Status in each model shows different leaders.
* Removing the leader shows leadership being claimed by another unit once the lease expires.

## Documentation changes

None

## Bug reference

None

### CODE REVIEW

File: core/raftlease/fsm.go [hunk 1]
```diff
@@ -583,9 +602,6 @@ func (c *Command) Validate() error {
 			return errors.NotValidf("setTime with model UUID")
 		}
 		if c.Lease != "" {
-			if c.Holder == "" {
```
Comment (howbazaar) [inline core/raftlease/fsm.go]: Why did you remove this?

File: core/raftlease/store_test.go [hunk 2]
```diff
@@ -356,6 +356,42 @@ func (s *storeSuite) TestLeasesFilter(c *gc.C) {
 	c.Check(s.fsm.Calls()[0].Args[1], jc.SameContents, []lease.Key{lease1, lease2})
 }
 
+func (s *storeSuite) TestLeaseGroup(c *gc.C) {
+	in5Seconds := s.clock.Now().Add(5 * time.Second)
```
Comment (howbazaar) [inline core/raftlease/store_test.go]: Is the clock a fake clock here?

File: core/raftlease/fsm.go (hunk 1 above)
Comment (babbageclunk) [reply core/raftlease/fsm.go]: I'm pretty sure it's a rogue cut'n'pasto - it doesn't make any sense to have a holder on a setTime. I've checked with @manadart but haven't heard back yet.

File: core/raftlease/store_test.go (hunk 2 above)
Comment (babbageclunk) [reply core/raftlease/store_test.go]: Yup!

**Comment by babbageclunk** [comment]
$$merge$$

**Comment by babbageclunk** [comment]
$$merge$$

**Comment by babbageclunk** [comment]
$$merge$$
